napari/layers/surface/surface.py

```diff
@@ -675,27 +683,18 @@ def _get_value_3d(
         # get the mesh triangles
         mesh_triangles = self._data_view[self._view_faces]
 
-        # get the triangles intersection
-        intersection_index, intersection = find_nearest_triangle_intersection(
-            ray_position=start_position,
-            ray_direction=ray_direction,
-            triangles=mesh_triangles,
-        )
-
-        if intersection_index is None:
-            return None, None
-
-        # add the full nD coords to intersection
-        intersection_point = start_point.copy()
-        intersection_point[dims_displayed] = intersection
+        intersection = intersect_line_with_triangles(
+            line_point=start_position,
+            line_direction=ray_direction,
+            triangles=np.array([mesh_triangles[self.hovered_face_index]]),
```

Comment:
There might be a small speedup to be had here by not creating a new array - something like this perhaps?
```suggestion
            triangles=mesh_triangles[None, self.hovered_face_index, :, :],  # `None` preserves the indexed axis
```